### PR TITLE
Fix bug of parsing of reactants in building the reaction SMILES

### PR DESCRIPTION
**Pull request by FanwangM**
The original codes would take anything other than REAGENT, SOLVENT, CATALYST, INTERNAL_STANDARD, PRODUCT as reactant, which is problematic. Because AUTHENTIC_STANDARD, BYPRODUCT, SIDE_PRODUCT are not reactants.

Also fixed the imports, duplicates and cleaning up.

### CODE REVIEW

File: ord_schema/message_helpers.py [hunk 1]
```diff
@@ -26,16 +26,15 @@
...
+from google.protobuf import json_format
 from rdkit import Chem
 from rdkit.Chem import rdChemReactions
 from werkzeug import security
 
 import ord_schema
 from ord_schema import units
-from ord_schema.proto import dataset_pb2
-from ord_schema.proto import reaction_pb2
+from ord_schema.proto import dataset_pb2, reaction_pb2
```
Comment (skearnes) [inline ord_schema/message_helpers.py:37]: generally when importing modules they should stay as separate imports

File: ord_schema/message_helpers.py [hunk 2]
```diff
@@ -739,7 +740,8 @@ def fetch_dataset(dataset_id: str, timeout: float = 10.0) -> dataset_pb2.Dataset
         RuntimeError: If the request fails.
         ValueError: If the dataset ID is invalid.
     """
-    from ord_schema import validations  # Avoid circular import; pylint: disable=import-outside-toplevel.
+    # Avoid circular import; pylint: disable=import-outside-toplevel.
```
Comment (skearnes) [inline ord_schema/message_helpers.py:743]: this comment needs to stay on the same line for the pylint flag to be localized

**Review by skearnes — CHANGES_REQUESTED** [review]

File: ord_schema/message_helpers.py (hunk 1 above)
Comment (FanwangM) [reply ord_schema/message_helpers.py]: Resolved in [d77d77a](https://github.com/open-reaction-database/ord-schema/pull/701/commits/d77d77a3a92e8dbc3b8d17b6e8633ed6a9d4e7f8)

File: ord_schema/message_helpers.py (hunk 2 above)
Comment (FanwangM) [reply ord_schema/message_helpers.py]: Resolved in [d77d77a](https://github.com/open-reaction-database/ord-schema/pull/701/commits/d77d77a3a92e8dbc3b8d17b6e8633ed6a9d4e7f8).

File: ord_schema/message_helpers_test.py [hunk 3]
```diff
@@ -18,13 +18,11 @@
 
 import pandas as pd
 import pytest
-from google.protobuf import json_format
-from google.protobuf import text_format
+from google.protobuf import json_format, text_format
```
Comment (skearnes) [inline ord_schema/message_helpers_test.py:21]: same comment here about importing modules on different lines

**Review by skearnes — APPROVED** [review]

File: ord_schema/message_helpers.py [hunk 4]
```diff
@@ -26,8 +26,8 @@
 import pandas as pd
 import requests
 from google import protobuf  # pytype: disable=import-error
-from google.protobuf import json_format
 from google.protobuf import text_format  # pytype: disable=import-error
+from google.protobuf import json_format
```
Comment (skearnes) [inline ord_schema/message_helpers.py:30]: keep these alphabetical (revert the change here)

Comment (FanwangM) [reply ord_schema/message_helpers.py]: Fixed in cf20d57.

File: ord_schema/message_helpers_test.py (hunk 3 above)
Comment (FanwangM) [reply ord_schema/message_helpers_test.py]: Fixed in cf20d57.
